### ISSUE TITLE
Removing invalid image tag Issue

### ISSUE THREAD

**Issue by lawrencealexander10** [dokku/dokku#5391]
### Description of problem

After creating a DO droplet with dokku and upgrading version to 0.28.1 I began following the application deployment tutorial. I'm using a dockerfile deployment. After attempting to push to dokku for the first time I encounter a "removing invalid image tag" error. I'm not sure if this has anything to do with upgrading to version 0.28.1.

<img width="565" alt="image" src="https://user-images.githubusercontent.com/7634661/194792294-fb8546bd-71a6-44ab-bf1c-18495593d6c6.png">

```
Enumerating objects: 2058, done.
Counting objects: 100% (2058/2058), done.
Delta compression using up to 8 threads
Compressing objects: 100% (886/886), done.
Writing objects: 100% (2058/2058), 783.09 KiB | 52.21 MiB/s, done.
Total 2058 (delta 1307), reused 1770 (delta 1117), pack-reused 0
remote: Resolving deltas: 100% (1307/1307), done.
-----> Cleaning up...
-----> Building tarot-backend from Dockerfile
dos2unix: Dockerfile: No such file or directory
dos2unix: Skipping Dockerfile, not a regular file.
remote:  !     Removing invalid image tag dokku/tarot-backend:latest
remote:  !     App build failed`
```

### dokku report $APP_NAME

----> uname: Linux dokku0214onubuntu2004-s-1vcpu-1gb-nyc1-01 5.4.0-52-generic #57-Ubuntu SMP Thu Oct 15 10:57:00 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux
-----> memory:
                     total        used        free      shared  buff/cache   available
       Mem:            981         295          86          12         599         516
       Swap:             0           0           0
-----> docker version:
       Client: Docker Engine - Community
        Version:           19.03.13
        API version:       1.40
        Go version:        go1.13.15
        Git commit:        4484c46d9d
        Built:             Wed Sep 16 17:02:52 2020
        OS/Arch:           linux/amd64
        Experimental:      false

       Server: Docker Engine - Community
        Engine:
         Version:          19.03.13
         API version:      1.40 (minimum version 1.12)
         Go version:       go1.13.15
         Git commit:       4484c46d9d
         Built:            Wed Sep 16 17:01:20 2020
         OS/Arch:          linux/amd64
         Experimental:     false
        containerd:
         Version:          1.6.8
         GitCommit:        9cd3357b7fd7218e4aec3eae239db1f68a5a6ec6
        runc:
         Version:          1.1.4
         GitCommit:        v1.1.4-0-g5fd4c4d
        docker-init:
         Version:          0.18.0
         GitCommit:        fec3683
-----> docker daemon info:
       Client:
        Debug Mode: true

       Server:
        Containers: 3
         Running: 3
         Paused: 0
         Stopped: 0
        Images: 10
        Server Version: 19.03.13
        Storage Driver: overlay2
         Backing Filesystem: extfs
         Supports d_type: true
         Native Overlay Diff: true
        Logging Driver: json-file
        Cgroup Driver: cgroupfs
        Plugins:
         Volume: local
         Network: bridge host ipvlan macvlan null overlay
         Log: awslogs fluentd gcplogs gelf journald json-file local logentries splunk syslog
        Swarm: inactive
        Runtimes: runc
        Default Runtime: runc
        Init Binary: docker-init
        containerd version: 9cd3357b7fd7218e4aec3eae239db1f68a5a6ec6
        runc version: v1.1.4-0-g5fd4c4d
        init version: fec3683
        Security Options:
         apparmor
         seccomp
          Profile: default
        Kernel Version: 5.4.0-52-generic
        Operating System: Ubuntu 20.04.5 LTS
        OSType: linux
        Architecture: x86_64
        CPUs: 1
        Total Memory: 981.2MiB
        Name: dokku0214onubuntu2004-s-1vcpu-1gb-nyc1-01
        ID: PEEB:CHZO:EMP7:XJR5:2ZB7:QM5F:TPDW:PHRP:MO73:X7YR:V5PN:LJDA
        Docker Root Dir: /var/lib/docker
        Debug Mode: false
        Registry: https://index.docker.io/v1/
        Labels:
        Experimental: false
        Insecure Registries:
         127.0.0.0/8
        Live Restore Enabled: false

-----> git version: git version 2.25.1
-----> sigil version: 0.9.0build+bc921b7
-----> herokuish version:
       herokuish: 0.5.37
       buildpacks:
         heroku-buildpack-multi     v1.2.0
         heroku-buildpack-ruby      v244
         heroku-buildpack-nodejs    v198
         heroku-buildpack-clojure   v90
         heroku-buildpack-python    v214
         heroku-buildpack-java      v72
         heroku-buildpack-gradle    v38
         heroku-buildpack-scala     v94
         heroku-buildpack-play      v26
         heroku-buildpack-php       v223
         heroku-buildpack-go        v166
         heroku-buildpack-nginx     v16
         buildpack-null             v3
-----> dokku version: dokku version 0.28.1
-----> plugn version: plugn: 0.12.0build+3a27594
-----> dokku plugins:
         00_dokku-standard    0.28.1 enabled    dokku core standard plugin
         20_events            0.28.1 enabled    dokku core events logging plugin
         app-json             0.28.1 enabled    dokku core app-json plugin
         apps                 0.28.1 enabled    dokku core apps plugin
         builder              0.28.1 enabled    dokku core builder plugin
         builder-dockerfile   0.28.1 enabled    dokku core builder-dockerfile plugin
         builder-herokuish    0.28.1 enabled    dokku core builder-herokuish plugin
         builder-lambda       0.27.0 enabled    dokku core builder-lambda plugin
         builder-null         0.28.1 enabled    dokku core builder-null plugin
         builder-pack         0.28.1 enabled    dokku core builder-pack plugin
         buildpacks           0.28.1 enabled    dokku core buildpacks plugin
         caddy-vhosts         0.28.1 enabled    dokku core caddy-vhosts plugin
         certs                0.28.1 enabled    dokku core certificate management plugin
         checks               0.28.1 enabled    dokku core checks plugin
         common               0.28.1 enabled    dokku core common plugin
         config               0.28.1 enabled    dokku core config plugin
         cron                 0.28.1 enabled    dokku core cron plugin
         docker-options       0.28.1 enabled    dokku core docker-options plugin
         domains              0.28.1 enabled    dokku core domains plugin
         enter                0.28.1 enabled    dokku core enter plugin
         git                  0.28.1 enabled    dokku core git plugin
         logs                 0.28.1 enabled    dokku core logs plugin
         mongo                1.21.0 enabled    dokku mongo service plugin
         network              0.28.1 enabled    dokku core network plugin
         nginx-vhosts         0.28.1 enabled    dokku core nginx-vhosts plugin
         plugin               0.28.1 enabled    dokku core plugin plugin
         postgres             1.24.0 enabled    dokku postgres service plugin
         proxy                0.28.1 enabled    dokku core proxy plugin
         ps                   0.28.1 enabled    dokku core ps plugin
         redis                1.24.0 enabled    dokku redis service plugin
         registry             0.28.1 enabled    dokku core registry plugin
         repo                 0.28.1 enabled    dokku core repo plugin
         resource             0.28.1 enabled    dokku core resource plugin
         run                  0.28.1 enabled    dokku core run plugin
         scheduler            0.28.1 enabled    dokku core scheduler plugin
         scheduler-docker-local 0.28.1 enabled    dokku core scheduler-docker-local plugin
         scheduler-null       0.28.1 enabled    dokku core scheduler-null plugin
         shell                0.28.1 enabled    dokku core shell plugin
         ssh-keys             0.28.1 enabled    dokku core ssh-keys plugin
         storage              0.28.1 enabled    dokku core storage plugin
         trace                0.28.1 enabled    dokku core trace plugin
         traefik-vhosts       0.28.1 enabled    dokku core traefik-vhosts plugin
=====> tarot-backend app-json information
       App json computed selected:    app.json
       App json global selected:      app.json
       App json selected:
=====> tarot-backend app information
       App created at:                1665365319
       App deploy source:
       App deploy source metadata:
       App dir:                       /home/dokku/tarot-backend
       App locked:                    false
=====> tarot-backend builder information
       Builder build dir:
       Builder computed build dir:
       Builder computed selected:     dockerfile
       Builder global build dir:
       Builder global selected:
       Builder selected:              dockerfile
=====> tarot-backend builder-dockerfile information
       Builder dockerfile computed dockerfile path: Dockerfile
       Builder dockerfile global dockerfile path: Dockerfile
       Builder dockerfile dockerfile path:
=====> tarot-backend builder-lambda information
       Builder lambda computed lambdayml path: lambda.yml
       Builder lambda global lambdayml path: lambda.yml
       Builder lambda lambdayml path:
=====> tarot-backend builder-pack information
       Builder pack computed projecttoml path: project.toml
       Builder pack global projecttoml path: project.toml
       Builder pack projecttoml path:
=====> tarot-backend buildpacks information
       Buildpacks computed stack:     gliderlabs/herokuish:latest-20
       Buildpacks global stack:
       Buildpacks list:
       Buildpacks stack:
=====> tarot-backend ssl information
       Ssl dir:                       /home/dokku/tarot-backend/tls
       Ssl enabled:                   false
       Ssl hostnames:
       Ssl expires at:
       Ssl issuer:
       Ssl starts at:
       Ssl subject:
       Ssl verified:
=====> tarot-backend checks information
       Checks disabled list:          none
       Checks skipped list:           none
=====> tarot-backend cron information
       Cron task count:               0
=====> tarot-backend docker options information
       Docker options build:          --file Dockerfile.deck_quart_webapi --link dokku.mongo.apimongo:dokku-mongo-apimongo --link dokku.postgres.apipostgres:dokku-postgres-apipostgres --link dokku.redis.apiredis:dokku-redis-apiredis
       Docker options deploy:         --link dokku.mongo.apimongo:dokku-mongo-apimongo --link dokku.postgres.apipostgres:dokku-postgres-apipostgres --link dokku.redis.apiredis:dokku-redis-apiredis --restart=on-failure:10
       Docker options run:            --link dokku.mongo.apimongo:dokku-mongo-apimongo --link dokku.postgres.apipostgres:dokku-postgres-apipostgres --link dokku.redis.apiredis:dokku-redis-apiredis
=====> tarot-backend domains information
       Domains app enabled:           true
       Domains app vhosts:            tarot-backend.dokku0214onubuntu2004-s-1vcpu-1gb-nyc1-01
       Domains global enabled:        true
       Domains global vhosts:         dokku0214onubuntu2004-s-1vcpu-1gb-nyc1-01
=====> tarot-backend git information
       Git deploy branch:             master
       Git global deploy branch:      master
       Git keep git dir:              false
       Git rev env var:               GIT_REV
       Git sha:
       Git last updated at:
=====> tarot-backend logs information
       Logs computed max size:        10m
       Logs global max size:          10m
       Logs global vector sink:
       Logs max size:
       Logs vector sink:
=====> tarot-backend network information
       Network attach post create:
       Network attach post deploy:
       Network bind all interfaces:          false
       Network computed attach post create:
       Network computed attach post deploy:
       Network computed bind all interfaces: false
       Network computed initial network:
       Network computed tld:
       Network global attach post create:
       Network global attach post deploy:
       Network global bind all interfaces:   false
       Network global initial network:
       Network global tld:
       Network initial network:
       Network static web listener:
       Network tld:
       Network web listeners:
=====> tarot-backend nginx information
       Nginx access log format:
       Nginx access log path:         /var/log/nginx/tarot-backend-access.log
       Nginx bind address ipv4:
       Nginx bind address ipv6:       ::
       Nginx client max body size:
       Nginx disable custom config:   false
       Nginx error log path:          /var/log/nginx/tarot-backend-error.log
       Nginx global hsts:             true
       Nginx computed hsts:           true
       Nginx hsts:
       Nginx hsts include subdomains: true
       Nginx hsts max age:            15724800
       Nginx hsts preload:            false
       Nginx proxy buffer size:       4096
       Nginx proxy buffering:         on
       Nginx proxy buffers:           8 4096
       Nginx proxy busy buffers size: 8192
       Nginx proxy read timeout:      60s
       Nginx last visited at:
       Nginx x forwarded for value:   $remote_addr
       Nginx x forwarded port value:  $server_port
       Nginx x forwarded proto value: $scheme
       Nginx x forwarded ssl:
=====> tarot-backend proxy information
       Proxy enabled:                 true
       Proxy port map:
       Proxy type:                    nginx
=====> tarot-backend ps information
       Deployed:                      false
       Processes:                     0
       Ps can scale:                  true
       Ps computed procfile path:     Procfile
       Ps global procfile path:       Procfile
       Ps procfile path:
       Ps restart policy:             on-failure:10
       Restore:                       true
       Running:                       false
=====> tarot-backend registry information
       Registry computed image repo:      dokku/tarot-backend
       Registry computed push on release: false
       Registry computed server:
       Registry global push on release:
       Registry global server:
       Registry image repo:
       Registry push on release:
       Registry server:
       Registry tag version:
=====> tarot-backend resource information
=====> tarot-backend scheduler information
       Scheduler computed selected:   docker-local
       Scheduler global selected:     docker-local
       Scheduler selected:
=====> tarot-backend scheduler-docker-local information
       Scheduler docker local disable chown:
       Scheduler docker local init process: true
       Scheduler docker local parallel schedule count:
=====> tarot-backend storage information
       Storage build mounts:
       Storage deploy mounts:
       Storage run mounts:

### Additional information

I am attempting to release a python flask like application

### Output of failing deploy after running: dokku trace:off

```shell
N/A
```

### Output of failing deploy after running: dokku trace:on

```shell
N/A UNable to push to begin first deploy
```

**Comment by josegonzalez** [comment]
Can you show the output of the following from the root of your repository?

```
ls -lah
```

**Comment by josegonzalez** [comment]
Also if you have a repo I can use to replicate this, that would be great.

**Comment by lawrencealexander10** [comment]
@josegonzalez Thanks for the quick response! I fixed the issue. I had been relying on the 0.21.0 version of the docs given the original dokku image for the DO droplet. What I did not realize was that after updating to 0.28.1 the `dokku docker-options:add ... build '--file ...'` command had been deprecated in 0.25.0. After unsetting the docker-options that I had placed and running the `dokku builder-dockerfile:set` command instead the invalid image warning went away and I was able to push.

Thanks again and appreciate your development of this project.
